### DISCUSSION TITLE
Check if ped gets hurt by player

### DISCUSSION THREAD

**Discussion by Dorbmon** [scripthookvdotnet/scripthookvdotnet#1309] [Q&A]
I want to check if the player hurts any ped no matter gun shoot or just drive a car and crash on the ped. Is there any way to do this? Thanks!

**Comment by Halen84** [comment]
See `ENTITY::HAS_ENTITY_BEEN_DAMAGED_BY_ENTITY`

**Comment by kagikn** [comment]
Use `Entity.HasBeenDamagedBy(Entity)` on peds other than the player ped, as Halen84 suggests a correct native for a solution. Or you can enumerate `EntityDamageRecordCollection` so you can fetch game time values when victims got hurt (the instance would enumerate the same struct that `ENTITY::HAS_ENTITY_BEEN_DAMAGED_BY_ENTITY` accesses).

Entity.HasBeenDamagedBy version:
```cs
Ped playerPed = Game.Player.Character;
/* bla bla */
bool hasVictimBeenDamagedByPlayer = victim.HasBeenDamagedBy(playerPed);
```

Entity.EntityDamageRecordCollection version:
```cs
Ped playerPed = Game.Player.Character;
/* bla bla */
bool hasVictimBeenDamagedByPlayer = victim.DamageRecords.Any(x => x.Attacker == playerPed); // eventually has the same meaning as ENTITY::HAS_ENTITY_BEEN_DAMAGED_BY_ENTITY, the native will also stops enumerating if one matching entry is found
```

Entity.EntityDamageRecordCollection version but you retrieve the entry if exists:
```cs
Ped playerPed = Game.Player.Character;
/* bla bla */
EntityDamageRecord damageRecord = victim.DamageRecords.FirstOrDefault(x => x.Attacker == playerPed); // eventually has the same meaning as ENTITY::HAS_ENTITY_BEEN_DAMAGED_BY_ENTITY, the native will also stops enumerating if one matching entry is found
if (damageRecord.WeaponHash != 0) /* looks weird, but weapon hash must not be zero, or the game will always treats invalid */
{
  // do something
}
```

**Reply by Dorbmon** [reply]
But if I want to check if the player drives a car hitting someone, how can I do that?

**Reply by kagikn** [reply]
You need to use `EntityDamageRecordCollection` to check if the player hits someone by car **without hitting false-positive cases**. Using `HasBeenDamagedBy(Entity)` in conjunction with `Entity.HasBeenDamagedBy(WeaponHash)` may result in a false-positive, since the underlying natives `HAS_ENTITY_BEEN_DAMAGED_BY_ENTITY` and `HAS_ENTITY_BEEN_DAMAGED_BY_WEAPON` enumerates all the damage records (up to 3).

Example:
```cs
Ped playerPed = Game.Player.Character;
WeaponHash runOverByCarWeaponHash = Game.GenerateHash("weapon_run_over_by_car"); // the ordinal one

/* bla bla */
EntityDamageRecord damageRecord = victim.DamageRecords.FirstOrDefault(x => x.Attacker == playerPed && x.WeaponHash == runOverByCarWeaponHash);
if (damageRecord.WeaponHash != 0) /* looks weird, but weapon hash must not be zero, or the game will always treats invalid */
{
  // do something
}
```
